### PR TITLE
fix(jans-auth-server): escaped postLogoutUri in EndSessionUtils

### LINKED ISSUE #14104: fix: fix(jans-auth-server): escaped postLogoutUri in EndSessionUtils -autocreated
This issue was created because no open issues were referenced in your PR (#14103).

### PR DESCRIPTION
**Pull request by yuriyz**
### Description

fix(jans-auth-server): escaped postLogoutUri in EndSessionUtils

#### Target issue
 
closes #14104  

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed logout page generation and strengthened security by properly escaping logout URIs and post-logout URLs in front-channel logout output, preventing injection via HTML attributes or embedded scripts.

* **Tests**
  * Added comprehensive tests covering logout HTML/JS escaping, edge-case inputs (quotes, script sequences) and ensuring state parameter handling does not duplicate existing values.

[Review Change Stack](https://app.coderabbit.ai/change-stack/JanssenProject/jans/pull/14103?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/76652c6c-0c37-40a6-b965-d1796b8d456a?refs=1a6b43883159d287939e232447b04fee55759006&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

**Comment by mo-auto** [comment]
Error: Hi @yuriyz, You did not reference an open issue in your [PR](#14103). I attempted to create an [issue](#14104) for you.
Please update that [issues'](#14104) title and body and make sure I correctly referenced it in the above PRs body.

File: jans-auth-server/server/src/main/java/io/jans/as/server/session/ws/rs/EndSessionUtils.java [hunk 1]
```diff
@@ -96,7 +98,7 @@ public static String createFronthannelHtml(Set<String> logoutUris, String postLo
 
             html += "<script>" +
                     "window.onload=function() {" +
-                    "window.location='" + postLogoutUrl + "'" +
+                    "window.location='" + StringEscapeUtils.escapeEcmaScript(postLogoutUrl) + "'" +
                     "}" +
                     "</script>";
```
[the remark this answers is not in the document]
Comment (yuriyz) [reply jans-auth-server/server/src/main/java/io/jans/as/server/session/ws/rs/EndSessionUtils.java:103]: @coderabbitai it's not clear what is the problem here. Please explain in details.

**Review by yuriyzz — APPROVED** [review]
